File: 3rd_task/3rd task.cpp
#include <iostream>
#include <vector>
#include <cmath>

using namespace std;

// Функция для проверки простоты числа
bool isProst(int num) {
    if (num <= 1) {
        return false;
    }
    for (int i = 2; i <= sqrt(num); i++) {
        if (num % i == 0) {
            return false;
        }
    }
    return true;
}

// Функция для подсчета количества простых чисел в векторе
int countProst(const vector<int>& numbers) {
    int count = 0;
    for (int num : numbers) {
        if (isProst(num)) {
            count++;
        }
    }
    return count;
}

int main() {
    setlocale(LC_ALL, "");
    int n;
    cout << "Введите количество чисел: ";
    cin >> n;

    vector<int> numbers(n);
    cout << "Введите " << n << " чисел: " << endl;
    for (int i = 0; i < n; i++) {
        cin >> numbers[i];
    }

    int result = countProst(numbers);
    cout << "Количество простых чисел: " << result << endl;

    return 0;
}

File: 3rd_task/3rd task.swift
import Foundation

// Функция для проверки простоты числа
func isProst(num: Int) -> Bool {
    if num <= 1 {
        return false
    }
    for i in 2...Int(sqrt(Double(num))) {
        if num % i == 0 {
            return false
        }
    }
    return true
}

// Функция для подсчета количества простых чисел в массиве
func countProst(numbers: [Int]) -> Int {
    var count = 0
    for num in numbers {
        if isProst(num: num) {
            count += 1
        }
    }
    return count
}

func main() {
    print("Введите количество чисел: ")
        guard let n = Int(readLine()!), n > 0 else {
        print("Неверное количество чисел")
        return
    }

    var numbers = [Int]()
    print("Введите \(n) чисел через Enter: ")
    for _ in 0..<n {
        guard let num = Int(readLine()!), num > 0 else {
            print("Неверное число")
            return
        }
        numbers.append(num)
    }

    let result = countProst(numbers: numbers)
    print("Количество простых чисел: \(result)")
}

main()
